src/main/scala/org/apache/sysml/api/dl/Utils.scala

```diff
@@ -111,6 +111,30 @@ object Utils {
 	  }
 	}
 	
+	class CopyCaffeDeconvFloatToSystemMLDeconvDoubleArray(data:java.util.List[java.lang.Float], F:Int, C:Int, H:Int, W:Int, arr:Array[Double]) 
+	    extends CopyFloatToDoubleArray(data, C, F*H*W, false, arr) {
+	  override def run(): Unit = {
```

Comment:
Also, in case it's helpful at any point, we added a DML function `transpose_NCHW_to_CNHW ` to transpose the first two dimensions: https://github.com/apache/systemml/blob/master/scripts/nn/util.dml#L223